__tests__/endpoints.test.js

```diff
@@ -15,26 +15,66 @@ afterAll(()=>{
-            .then(({body})=>{
-                expect(Array.isArray(body.topics)).toBe(true)
-                expect(body.topics.length).toBe(3)
-                body.topics.forEach(topic=>{
-                    expect(typeof topic.description).toBe('string')
-                    expect(typeof topic.slug).toBe('string')
+        describe('GETting all topics',()=>{
+            test('GET /api/topics',()=>{
+                return request(app)
+                .get('/api/topics')
+                .expect(200)
+                .then(({body})=>{
+                    expect(Array.isArray(body.topics)).toBe(true)
+                    expect(body.topics.length).toBe(3)
+                    body.topics.forEach(topic=>{
+                        expect(typeof topic.description).toBe('string')
+                        expect(typeof topic.slug).toBe('string')
+                    })
                 })
             })
         })
-        test('GET /api', ()=>{
-            return request(app)
-            .get('/api')
-            .expect(200)
-            .then(({body})=>{
-                expect(body.endpoints).toEqual(endpointsObj)
+        describe('GETting all the available apis',()=>{
+            test('GET /api', ()=>{
+                return request(app)
+                .get('/api')
+                .expect(200)
+                .then(({body})=>{
+                    expect(body.endpoints).toEqual(endpointsObj)
+                })
+            })
+        })
+
+        describe('GETting article by article id',()=>{
+            test('GET /api/articles/:article_id', ()=>{
+                return request(app)
+                .get('/api/articles/1')
+                .expect(200)
+                .then(({body})=>{
+                    expect(body.article.article_id).toBe(1);
+                    expect(typeof body.article.title).toBe('string');
+                    expect(typeof body.article.topic).toBe('string');
+                    expect(typeof body.article.author).toBe('string');
+                    expect(typeof body.article.body).toBe('string');
+                    expect(typeof body.article.created_at).toBe('string');
+                    expect(typeof body.article.votes).toBe('number');
+                    expect(typeof body.article.article_img_url).toBe('string');
+    
+                })
+            })
+            test('Respond with 400 when given invalid param',()=>{
+                return request(app)
+                .get('/api/articles/one')
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe("Bad Request")
+                })
+            })
+            test('Respond with 404 when given valid but non existent param',()=>{
+                return request(app)
+                .get('/api/articles/4000')
+                .expect(404)
+                .then(({body})=>{
+                    expect(body.msg).toBe("article does not exist")
+                })
```

Comment:
Great sad path tests 😄 I'd consider making the test descriptions more specific, for example: `"should respond with a status 404 and error message if endpoint is a valid but non-existent article id"`